client/components/number-field/number-input.js

```diff
@@ -1,44 +1,38 @@
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import FormTextInput from 'components/forms/form-text-input';
 
-export default React.createClass( {
-	displayName: 'NumberInput',
-
-	propTypes: {
+export default class NumberInput extends Component {
+	static propTypes = {
 		onChange: PropTypes.func,
-	},
+	};
 
-	getDefaultProps() {
-		return {
-			onChange: () => {},
-		};
-	},
+	static defaultProps = {
+		onChange: () => {},
+	};
 
-	getInitialState() {
-		return {
-			focused: false,
-			text: this.props.value,
-		};
-	},
+	state = {
+		focused: false,
+		text: this.props.value,
+	};
 
 	componentWillReceiveProps( nextProps ) {
 		if ( ! this.state.focused && nextProps.value !== this.props.value ) {
 			this.setState( { text: nextProps.value } );
 		}
-	},
+	}
 
-	handleChange( event ) {
+	handleChange = ( event ) => {
```

Comment:
Definitely! Arrow functions' `this` value is always set to the value of `this` of their enclosing scope. In this case, it's referring to an instance of `<NumberField />`. For this reason, we don't need to also bind handleChange to `this` in the constructor.

This syntax is widely used in Calypso, and also listed/explained in the [above reference](https://babeljs.io/blog/2015/06/07/react-on-es6-plus#arrow-functions) to React on ES6+.